### PR TITLE
Ritika/bac 129 close position

### PR DESCRIPTION
**Pull request by Ritika-Agrawal811**

### CODE REVIEW

File: internal/api/routes/user_route.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ func setupUserRoutes(e *echo.Echo, services *ServiceContainer) {
 	userGroup.GET("/referral-link", userHandler.GetReferralLinkHandler)
 	userGroup.GET("/reward", userHandler.GetRewardPointsByWallet)
 	userGroup.POST("/order", userHandler.AddOrderHandler)
+	userGroup.POST("/position/close", userHandler.StorePositionHandler)
```
Comment (gopherine) [inline internal/api/routes/user_route.go]: Naming can be correct if its close position then it should be close position

also i think this should be placed in its own route file

positions route and all the routes should be placed there i dont think so this is part of the user group

File: internal/domain/dtos/user.go [hunk 2]
```diff
@@ -55,3 +55,30 @@ type UserNonce struct {
...
+	Wallet          string  `json:"wallet" validate:"required"`
+	Quantity        float64 `json:"quantity" validate:"required"`
+	Side            string  `json:"side" validate:"required"`
+	Asset           string  `json:"asset" validate:"required"`
+	Price           float64 `json:"price" validate:"required"`
+	ExpireTime      string  `json:"expire_time" validate:"required"`
+	OptionType      string  `json:"option_type" validate:"required"`
+	Status          string  `json:"status" validate:"required"`
+	TransactionTime string  `json:"transaction_time" validate:"required"`
+}
```
Comment (gopherine) [inline internal/domain/dtos/user.go]: I dont see any identifier here that says its specific to 1 user
i believe it is generic hence again can be renamed to  OrderDetails

File: internal/domain/entities/options_data.go [hunk 3]
```diff
@@ -131,15 +131,18 @@ type OrderbookResponse struct {
...
+	EntryPrice    float64     `json:"entry_price"`
+	ROI           float64     `json:"roi"`
+	Position      string      `json:"position"`
+	ExpiryDate    string      `json:"expiry_date"`
+	Side          string      `json:"side"`
+	Pnl           float64     `json:"pnl"`
+	Symbol        string      `json:"symbol"`
+	OptionType    string      `json:"option_type"`
+	EntryOrderIds interface{} `json:"entry_order_ids"`
 }
```
Comment (gopherine) [inline internal/domain/entities/options_data.go]: Should be renamed to position

File: internal/domain/entities/options_data.go [hunk 4]
```diff
@@ -131,15 +131,18 @@ type OrderbookResponse struct {
...
 
 type UserOrdersResponse struct {
```
Comment (gopherine) [inline internal/domain/entities/options_data.go]: should be renamed to OrderResponse

**Review by gopherine — APPROVED** [review]

File: internal/domain/dtos/user.go (hunk 2 above)
Comment (Ritika-Agrawal811) [reply internal/domain/dtos/user.go]: it has wallet which is for a particular user only

**Review by gopherine — APPROVED** [review]

**Pull request merged**